src/highlighter.js

```diff
@@ -136,7 +165,7 @@ export default function (options) {
                         startNodeContainer: range.startContainer,
                         startOffset: range.startOffset,
 
-                        endNode: isElement(range.endContainer)
+                        endNode: isElement(range.endContainer) && range.endOffset > 0
```

Comment:
FYI: If you do a double-click, one word is selected. If you do a triple-click, whole element is selected. This selection is a bit weird: for
```
  <div>
    <h2>here</h2>  <!-- triple-click here -->
    <p>there</p>
  </div>
```
range will be 
```
commonAncestorContainer: '<div>' element node
startContainer: 'here' text node
startOffset: 0
endContainer: '<p>' element node
endOffset: 0
```

Such selection isn't handled well by existing code - it continues to highlight until the end of `range.commonAncestorContainer`
https://screenrec.com/share/60Fl9QfbhM
https://screenrec.com/share/Ycn24KzAqG

Same happens not only on triple-click, but also for some mouse positions, where you mouseup below your element.